kubernetes/metadata/resource.go

```diff
@@ -167,23 +172,27 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
 }
 
 func generateMapSubset(input map[string]string, keys []string, dedot bool) mapstr.M {
+	var matched bool
 	output := mapstr.M{}
 	if input == nil {
 		return output
 	}
 
 	for _, key := range keys {
-		value, ok := input[key]
-		if ok {
-			if dedot {
-				dedotKey := utils.DeDot(key)
-				_, _ = output.Put(dedotKey, value)
-			} else {
-				_ = safemapstr.Put(output, key, value)
+		for label, value := range input {
+			matched, _ = regexp.MatchString(key, label)
+			if matched == true {
+				if dedot {
+					dedotKey := utils.DeDot(label)
+					_, _ = output.Put(dedotKey, value)
+				} else {
+					_ = safemapstr.Put(output, label, value)
+				}
+
 			}
+			matched = false
```

Comment:
nit: maybe set it right before the `Match` at line 183 so as to be more explicit and clear?

Reply:
Crossing your comment with the update. Fixed